packages/sync/src/cache/createSubscriptionOptions.ts

```diff
@@ -0,0 +1,94 @@
+import { SubscribeToMoreOptions, OperationVariables, ObservableQuery } from "apollo-client";
+import { QueryWithVariables, UpdateFunction, Query } from "./CacheUpdates";
+import { DocumentNode } from "graphql";
+import { CacheOperation } from "./CacheOperation";
+import { getOperationFieldName } from "..";
+
+export interface SubscriptionHelperOptions {
+  subscriptionQuery: Query;
+  cacheUpdateQuery: Query;
+  operationType: CacheOperation;
+  idField?: string;
+}
+
+export const subscribeToMoreHelper = (observableQuery: ObservableQuery,
+                                      arrayOfHelperOptions: SubscriptionHelperOptions[]) => {
+  for (const option of arrayOfHelperOptions) {
+    observableQuery.subscribeToMore(createSubscriptionOptions(option));
+  }
+};
+
+export const createSubscriptionOptions = (options: SubscriptionHelperOptions): SubscribeToMoreOptions => {
+  const {
+    subscriptionQuery,
+    cacheUpdateQuery,
+    operationType,
+    idField = "id"
+  } = options;
+  const document = (subscriptionQuery && (subscriptionQuery as QueryWithVariables).query)
+    || (subscriptionQuery as DocumentNode);
+  const variables = (subscriptionQuery && (subscriptionQuery as QueryWithVariables).variables)
+    || {} as OperationVariables;
+  const query = (cacheUpdateQuery && (cacheUpdateQuery as QueryWithVariables).query)
+    || (cacheUpdateQuery as DocumentNode);
+  const queryField = getOperationFieldName(query);
+
+  return {
+    document,
+    variables,
+    updateQuery: (prev, { subscriptionData }) => {
+      const data = subscriptionData.data;
+      const [key] = Object.keys(data);
+      const mutadedItem = data[key];
+
+      const optype = operationType;
+      const obj = prev[queryField];
+
+      const updater = getUpdateQueryFunction(optype, idField);
+      const result = updater(obj, mutadedItem);
+      return {
+        [queryField]: result
+      };
+    }
+  };
+};
+
+export const getUpdateQueryFunction = (opType: CacheOperation, idField = "id"): UpdateFunction => {
+  let updateFunction: UpdateFunction;
+
+  switch (opType) {
+    case CacheOperation.ADD:
+      updateFunction = (prev, newItem) => {
+        if (!newItem) {
+          return [...prev];
+        } else {
+          return [...prev.filter(item => {
+            return item[idField] !== newItem[idField];
+          }), newItem];
+        }
+      };
+      break;
+    case CacheOperation.REFRESH:
```

Comment:
[Question] So for subscriptions we need refresh method?
What I'm worried that cache operations are common for mutation and subscriptions so people may pass refresh to mutate etc. Let's chat about options here before we apply anything because it may be not worth it.

Reply:
Yeah we need refresh for subscriptions. I've thought of that. I pass update but if you look here: https://github.com/aerogear/aerogear-js-sdk/blob/cache-update-helpers/packages/sync/src/cache/createMutationOptions.ts#L98 I have a default which returns no update function for an update.

I use it in the showcase here to make the point that it doesn't break anything: https://github.com/aerogear/ionic-showcase/blob/breakup-subscriptions/src/app/services/sync/item.service.ts#L78

Reply:
@StephenCoady - Let's update comment that refresh is not default and it is used for subscriptions only then. OR rename `Subscription_UPDATE` ?